src/main/java/io/aiven/kafka/connect/common/output/jsonwriter/JsonLinesOutputWriter.java

```diff
@@ -51,12 +48,17 @@
         objectMapper.setNodeFactory(JsonNodeFactory.withExactBigDecimals(true));
     }
 
+    @Override
+    public void startWriting(final OutputStream outputStream) throws IOException {
```

Comment:
I wonder: maybe the start of writing can be handled by the first call to `writeRecord`? It's a matter of a boolean field.

Reply:
Same here. `JsonLinesOutputWriter` is thread safe ... maybe it would be better to keep boolean flag or counter (e.g. for statistics) inside the flag to check is it the first record or not or even better check the size of the map `Map::isEmpty`

Reply:
My Scala background always push me towards stateless solutions. Also, there were a `writeLastRecord`, so adding something that add a first one looks logical: in this case Writer remains `passive` and just waits for a requests outside.

Nevertheless, we could refactor `OutputWriter` interface a bit and  and make it responsible for a single OutputStream:
make it an abstract class with a Stream Object in constructor. In this case, exactly one stream connected to one writer so it is safe to make `startWriting` functionality during initialization. Also, in this case would not be necessary to send an outputStream as a parameter.